robot-core/src/main/java/org/obolibrary/robot/OntologyHelper.java

```diff
@@ -50,6 +50,71 @@ public class OntologyHelper {
   private static final String nullIRIError =
       NS + "NULL IRI ERROR import ontology does not have an IRI";
 
+  /**
+   * Given an ontology, an entity, annotates the entity with the given annotation in the ontology.
+   *
+   * @param ontology the ontology to modify
+   * @param owlEntity the entity to annotate
+   * @param property annotation property
+   * @param value the IRI or literal value to add
+   * @param overload when false: if the entity already has an annotation with the given property,
+   *     doesn't add the annotation. When true, adds the annotation anyway.
+   */
+  public static void addEntityAnnotation(
+      OWLOntology ontology,
+      OWLEntity owlEntity,
+      OWLAnnotationProperty property,
+      OWLAnnotationValue value,
+      boolean overload) {
```

Comment:
`overload` does not seem to be used. I can see the benefit of including it, but if `overload` is `true` then it's a waste to look for an `existingAnnotation`.

Reply:
You rigour is insane - I would have never seen this. I added another conditional, there is probably a more elegant programmatic solution, but I hope it satisfies your concern?